internal/planner/planner.go

```diff
@@ -286,6 +286,11 @@ func findFilteredByRelationFields(
 }
 
 func (p *Planner) tryOptimizeJoinDirection(node *invertibleTypeJoin, parentPlan *selectTopNode) error {
+	if !node.childSide.relFieldDef.HasValue() {
+		// If the relation is one sided we cannot invert the join, so return early
+		return nil
```

Comment:
suggestion: If not too much trouble would be nice to hit this case as it shows not covered right now.

Reply:
It might be dead code, but I'm not sure - the invertable join stuff is quite complex and if it isn't hit by the new test I don't know what will hit it.  And I don't want to spend much time on this right now, especially given the pending secondary relation rework Fred mentioned.